samples/bluetooth/ble_bms/src/main.c

```diff
@@ -0,0 +1,745 @@
+ */
+
+#include <stdint.h>
+#include <hal/nrf_gpio.h>
+#include <nrf_error.h>
+
+#include <nrf_soc.h>
+#include <ble_gap.h>
+#include <ble_types.h>
+#include <bm/bm_timer.h>
+#include <bm/bm_buttons.h>
+#include <bm/bluetooth/ble_adv.h>
+#include <bm/bluetooth/ble_qwr.h>
+#include <bm/bluetooth/services/common.h>
+#include <bm/bluetooth/services/uuid.h>
+#include <bm/bluetooth/services/ble_bms.h>
+#include <bm/bluetooth/services/ble_dis.h>
+#include <bm/bluetooth/peer_manager/nrf_ble_lesc.h>
+#include <bm/bluetooth/peer_manager/peer_manager.h>
+#include <bm/bluetooth/peer_manager/peer_manager_handler.h>
+#include <bm/softdevice_handler/nrf_sdh.h>
+#include <bm/softdevice_handler/nrf_sdh_ble.h>
+
+#include <zephyr/sys/ring_buffer.h>
+#include <zephyr/toolchain.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_ctrl.h>
+#include <board-config.h>
+
+/* FIFO for keeping track of keystrokes that can not be sent immediately. */
+RING_BUF_DECLARE(peers_to_delete_on_disconnect,
+		 CONFIG_APP_BLE_BMS_PEERS_TO_DELETE_ON_DISCONNECT_MAX * sizeof(uint16_t));
+
+LOG_MODULE_REGISTER(app, CONFIG_APP_BLE_BMS_LOG_LEVEL);
+
+/* Perform bonding. */
+#define SEC_PARAM_BOND 1
+/* Man In The Middle protection not required. */
+#define SEC_PARAM_MITM 0
+/* LE Secure Connections enabled. */
+#define SEC_PARAM_LESC 1
+/* Keypress notifications not enabled. */
+#define SEC_PARAM_KEYPRESS 1
+/* No I/O capabilities. */
+#define SEC_PARAM_IO_CAPABILITIES BLE_GAP_IO_CAPS_DISPLAY_YESNO
+/* Out Of Band data not available. */
+#define SEC_PARAM_OOB 0
+/* Minimum encryption key size. */
+#define SEC_PARAM_MIN_KEY_SIZE 7
+/* Maximum encryption key size. */
+#define SEC_PARAM_MAX_KEY_SIZE 16
+
+/* BLE Advertising library instance */
+BLE_ADV_DEF(ble_adv);
+/* BLE QWR instance */
+BLE_QWR_DEF(ble_qwr);
+/* BLE Bond Management Service instance */
+BLE_BMS_DEF(ble_bms);
+
+/* BLE Connection handle */
+static uint16_t conn_handle = BLE_CONN_HANDLE_INVALID;
+/* Peer ID */
+static uint16_t peer_id;
+/* Flag for ongoing authentication request */
+static bool auth_key_request;
+
+/* Write buffer for the Queued Write module. */
+static uint8_t qwr_mem[CONFIG_APP_QWR_MEM_BUFF_SIZE];
+
+/* Forward declaration */
+static void identities_set(enum pm_peer_id_list_skip skip);
+
+static void delete_disconnected_bonds(void);
+
+static void on_ble_evt(const ble_evt_t *evt, void *ctx)
+{
+	uint32_t nrf_err;
+
+	switch (evt->header.evt_id) {
+	case BLE_GAP_EVT_CONNECTED:
+		LOG_INF("Peer connected, conn handle %d", evt->evt.gap_evt.conn_handle);
+		conn_handle = evt->evt.gap_evt.conn_handle;
+
+		nrf_err = ble_qwr_conn_handle_assign(&ble_qwr, conn_handle);
+		if (nrf_err) {
+			LOG_ERR("Failed to assign qwr handle, nrf_error %#x", nrf_err);
+			return;
+		}
+		nrf_gpio_pin_write(BOARD_PIN_LED_0, !BOARD_LED_ACTIVE_STATE);
+		nrf_gpio_pin_write(BOARD_PIN_LED_1, BOARD_LED_ACTIVE_STATE);
+		break;
+
+	case BLE_GAP_EVT_DISCONNECTED:
+		LOG_INF("Peer disconnected, reason %d",
```

Comment:
As far as I can see, these values (`BLE_HCI_STATUS_CODES`) are defined as hex.
```suggestion
		LOG_INF("Peer disconnected, reason %#x",
```